src/Zio/FileSystems/PhysicalFileSystem.cs

```diff
@@ -815,9 +815,7 @@ protected override string ConvertPathToInternalImpl(UPath path)
             if (absolutePath.Length > DrivePrefixOnWindows.Length + 1)
                 builder.Append(absolutePath.Replace(UPath.DirectorySeparator, '\\').Substring(DrivePrefixOnWindows.Length + 2));
 
-            var result = builder.ToString();
-            builder.Length = 0;
```

Comment:
To simplify / alingn the pattern, I've made it so that when obtained, the length get set, so you know it's good to use. 

I removed the reset of the length when used, as it was not applied consistently nor was the exception handling equivalent everywhere.